pages/api/datarooms/paged/index.ts

```diff
@@ -0,0 +1,137 @@
+    const { id } = req.query as { id: string };
+    try {
+      const dataroom = await prisma.dataroom.findUnique({
+        where: {
+          id,
+        },
+        include: {
+          files: true,
+        },
+      });
+      if (!dataroom) {
+        return res.status(404).end("Dataroom not found");
+      }
+      return res.status(200).json(dataroom);
+    } catch (error) {
+      return res.status(500).json({
+        message: "Internal Server Error",
+        error: (error as Error).message,
+      });
+    }
+  } else if (req.method === "POST") {
+    // POST /api/datarooms/paged
+    const session = await getServerSession(req, res, authOptions);
+    if (!session) {
+      res.status(401).end("Unauthorized");
+      return;
+    }
+
+    //Input validation
+    let name: string;
+    let description: string;
+    let titles: string[];
+    let links: string[];
+    let password: string;
+    let emailProtected: boolean;
+    let teamId: string;
+    const userId = (session?.user as CustomUser).id;
+
+    try {
+      //Input Validation
+      ({ name, description, titles, links, password, emailProtected, teamId } =
+        bodySchema.parse(req.body));
+      //Check if user if member of team
+      await isUserMemberOfTeam({ teamId, userId });
+      const dataroomName = await prisma.dataroom.findFirst({
+        where: {
+          name: name,
+        },
+      });
+
+      if (dataroomName) {
+        return res.status(409).json({
+          message: `A dataroom with name "${name}" already exists. Please try another name`,
+        });
+      }
+
+      // Create dataroom
+      const dataroom = await prisma.dataroom.create({
+        data: {
+          name: name,
+          description: description,
+          type: "PAGED",
+          emailProtected: emailProtected,
+          password: password,
+          ownerId: (session.user as CustomUser).id,
+          teamId,
+        },
+      });
+
+      //Save files to dataroom
+      for (let i = 0; i < links.length; i++) {
+        await prisma.dataroomFile.create({
+          data: {
+            name: titles[i],
+            url: links[i],
+            dataroomId: dataroom.id,
+          },
+        });
+      }
+
+      await identifyUser((session.user as CustomUser).id);
+      await trackAnalytics({
+        event: "Dataroom Created",
+        dataroomId: dataroom.id,
+        name: dataroom.name,
+      });
+
+      res.status(201).json(dataroom);
+    } catch (error) {
+      if (error instanceof TeamError) {
+        return res.status(401).json({ message: "Unauthorized access" });
+      } else if (error instanceof ZodError) {
+        return res.status(403).json({
+          message: "Invalid Inputs",
+          error: (error as Error).message,
+        });
+      }
+      log(`Failed to create dataroom. Error: \n\n ${error}`);
+      res.status(500).json({
+        message: "Internal Server Error",
+        error: (error as Error).message,
+      });
+    }
```

Comment:
you can use `errorHandler(error, res)` for the TeamError and `500` errors, also you can add the new error class of ZodError inside the `errorHandler` function and clean up this catch block.

Reply:
Done